cmd/vicadmin/server.go

```diff
@@ -499,12 +507,41 @@ func (s *server) tailFiles(res http.ResponseWriter, req *http.Request, names []s
 	}
 }
 
+// deriveDisplayError takes a vSphere session error and returns
+// an error string that is safe to display on the vicadmin page.
+func deriveDisplayError(err error) string {
+	var displayErrStr string
+
+	if err != nil {
+		switch err := err.(type) {
+		case session.SDKURLError:
+			displayErrStr = fmt.Sprintf("SDK URL could not be parsed: %s", err.Err)
+		case session.SoapClientError:
+			displayErrStr = "unable to obtain a vim client"
+		case session.UserPassLoginError:
+			displayErrStr = "unable to log in with username and password"
+		default:
+			displayErrStr = err.Error()
+		}
+	}
+
+	return displayErrStr
+}
```

Comment:
IMHO - it's more confusing this way. You are hiding the fact that you are returning an empty string when err == nil

```
func deriveDisplayError(err error) string {
	if err != nil {
		switch err := err.(type) {
		case session.SDKURLError:
			return fmt.Sprintf("SDK URL could not be parsed: %s", err.Err)
 		case session.SoapClientError:
 			return "unable to obtain a vim client"
 		case session.UserPassLoginError:
 			return "unable to log in with username and password"
 		default:
 			return err.Error()
 		}
 	}
 
 	return ""
 }
```